src/pages/documentation/mobile-core/tabs/api-reference.md

```diff
@@ -390,6 +513,117 @@ const DEBUG = "ACP_LOG_LEVEL_DEBUG";
 const VERBOSE = "ACP_LOG_LEVEL_VERBOSE";
 ``` --->
 
+<Variant platform="android" api="register-event-listener" repeat="5"/>
+
+#### Java
+
+**Syntax**
+
+```java
+public static void registerEventListener(@NonNull final String eventType, @NonNull final String eventSource, @NonNull final AdobeCallback<Event> callback)
+```
+
+**Example**
+
+```java
+MobileCore.registerEventListener(EventType.CONFIGURATION, EventSource.RESPONSE_CONTENT, new AdobeCallback<Event>() {
+    @Override
+    public void call(Event value) {
+        // handle event
+    }
+});
+```
+
+<Variant platform="ios" api="register-event-listener" repeat="10"/>
+
+
+#### Swift
+
+**Syntax**
+
+```swift
+public static func registerEventListener(type: String, source: String, listener: @escaping EventListener)
+```
+
+**Example**
+
+```swift
+MobileCore.registerEventListener(type: EventType.configuration, source: EventSource.responseContent, listener: { event in
+   // handle event 
+})
+```
+
+#### Objective-C
+
+**Syntax**
+
+```objc
+@objc(registerEventListenerWithType:source:listener:)
```

Comment:
This is not a true objc syntax. Added these in this style because the rest of the ObjC syntaxes are in the same form. Discussed with @praveek that these may need to be changed when consensus is achieved on consistency for  syntax for obj versions across all docs.